dpypelines/pipeline/shared/transforms/sdmx/compact/v20/prototype/v1.py

```diff
@@ -1,99 +1,68 @@
-            if i[0]!='na_:Obs':
-                headers.append(i[0])
+    tree = ET.parse(input_path)
+    root = tree.getroot()
 
-    output = []
+    series_dict = {}
+    obs_dict = {}
+    for series in root.iter():
+        if "Series" in series.tag:
+            for obs in series.iter():
+                if "Obs" in obs.tag:
+                    convert(series.attrib, series_dict)
+                    convert(obs.attrib, obs_dict)
 
-    # and this is the main loop which goes through each series block and pulls out the observational data and sticks in under the right header where it exists, then adding the new columns for the remaining variables inside the obs tags
+    series_frame = pd.DataFrame(series_dict)
+    obs_frame = pd.DataFrame(obs_dict)
 
-    for table_index in range(len(tables)):
-        list = tables[table_index] # get a series block
-        headers_df = header_frame # create an df of the headers we have so far
-        obs_df = pd.DataFrame() # create an empty df for the headers remaining in the obs tag
+    header_dict = flatten_dict(header)
+    header_df = pd.DataFrame([header_dict])
 
-        for item in list.items(): # as we move through the series block we go through each value and see if it matches one of the headers we already have
-            if item[0] in headers: # if the header matches we insert the value for this header
-                temp = []
-                temp.append(item[1])
-                headers_df[item[0]] = temp
-                # I think this is a point of performance loss, we should only have to check the headers once per series block rather than looping past it every time (TODO)
-            else: # if not then its an observation so we crete a temporary mini df with the 4 values in the obs tag and then append it to a dataframe containing all the observations for this series block, then repeat for each obs tag
-                for entry in item[1]:
-                    temp_df = pd.DataFrame()
-                    for obs in entry.items():
-                        temp = []
-                        temp.append(obs[1])
-                        temp_df[obs[0]] = temp
-                    obs_df = pd.concat([obs_df,temp_df])
-            
-        table_df = pd.concat([headers_df, obs_df], axis = 1) # merge the headers to each observation
+    # Here the records on the header dataframe are replicated to match the length of the series and observation dataframes.
+    repl_rows = header_df.loc[0].copy()
+    header_frame = header_df._append(
+        [repl_rows] * (len(obs_frame) - 1), ignore_index=True
+    )
 
-        output.append(table_df) # add to a list of all the series blocks that have been processed 
-
-    full_table = pd.concat(output) # merge all the series blocks together
-
-    # the following is just tidying up the column headers so theyre not filled with @ and such
-
-    header_replace = { x: str(x).replace('@', '') for x in full_table.columns }
```

Comment:
I think these 2 rows should be left in for consistency (removing the @'s from the column names)

Reply:
Absolutely! I missed the two lines of codes.
I have included them.